tfx_addons/copy_example_gen/component.py

```diff
@@ -53,42 +55,82 @@ def CopyExampleGen(  # pylint: disable=C0103
+  `split_name`. Following the creation of the `Split-{name}` folder, the files
+  in the URI path will then be copied into the designated `Split-{name}` folder.
+
+  Args:
+    input_json_str: JSON string containing the split labels (key) and URIs
+      containing the split label's TFRecords (value). These TFRecords are copied
+      to `output_example`.
+    output_example: Output Examples object containing the output `uri` and
+      `split_names`. This value specifies an OutputArtifact, and does not need
+      to be provided by the caller.
   """
+  logging.getLogger().setLevel(logging.INFO)
+  input_dict = create_input_dictionary(input_json_str)
 
-  # Convert primitive type str to Dict[str, str].
-  input_dict = json.loads(input_json_str)
-
-  # Creates directories from the split-names and tfrecord uris provided into
-  # output_example.split_names property.
-  tfrecords_list = []
   output_example_uri = output_example.uri
 
   for split_label, split_tfrecords_uri in input_dict.items():
     # Create Split-name folder name and create directory.
-    # output_example_uri = output_example.uri
-    split_value = (f"/Split-{split_label}/")
-    fileio.mkdir(f"{output_example_uri}{split_value}")
-
-    # Pull all files from uri.
-    tfrecords_list = fileio.glob(f"{split_tfrecords_uri}*.gz")
+    split_value_uri = f"{output_example_uri}/Split-{split_label}/"
+    fileio.mkdir(f"{split_value_uri}")
 
-    # Copy files into folder directories.
-    for tfrecord in tfrecords_list:
-      file_name = os.path.basename(os.path.normpath(tfrecord))
-      file_destination = (f"{output_example_uri}{split_value}{file_name}")
-      fileio.copy(tfrecord, file_destination, True)
+    copy_examples(split_tfrecords_uri, split_value_uri)
 
   # Build split_names in required Examples Artifact properties format.
   example_properties_split_names = "[\"{}\"]".format('","'.join(
       input_dict.keys()))
   output_example.split_names = example_properties_split_names
+
+
+def create_input_dictionary(input_json_str: str) -> Dict[str, str]:
+  """Creates a dictionary from input JSON string.
+
+  Args:
+    input_json_str: JSON string with Split label (key) to Split URI (value).
+  """
+  # Convert primitive type str to Dict[str, str].
+  if len(input_json_str) == 0:
+    raise ValueError(
+        "Input string is not provided. Expected format is Split label (key) "
+        "and Split URI (value).")
+
+  input_dict = json.loads(input_json_str)
+  if not isinstance(input_dict, dict):
+    raise ValueError(
+        f"Input string {input_json_str} is not provided as a dictionary. "
+        "Expected format is Split label (key) and Split URI (value).")
+  if len(input_dict.items()) == 0:
+    raise ValueError(
+        "Input dictionary is empty. Expected format is Split label (key) "
+        "and Split URI (value).")
+  return input_dict
+
+
+def copy_examples(split_tfrecords_uri: str, split_value_uri: str) -> None:
```

Comment:
It's used directly in the tests, but if we want to make it private I could add a pylint exception. what do you think?

Reply:
in general I would recommend to keep everything private except for whatever users will use. If we need to add pylint exception let's do that.

The reason is that if a user was now to use your functions and we would need to change them bc whatever reason, this could lead to us breaking backwards compatibility. By making them private we say to users that we can change it whenever.

Reply:
ack, modified to a private function